conans/test/functional/package_id/compatible_test.py

```diff
@@ -0,0 +1,117 @@
+import unittest
+
+from conans.test.utils.tools import TestClient, GenConanfile
+
+
+class CompatibleIDsTest(unittest.TestCase):
+
+    def compatible_setting_test(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile, CompatiblePackage
+
+            class Pkg(ConanFile):
+                settings = "os", "compiler"
+                def package_id(self):
+                    if self.settings.compiler == "gcc" and self.settings.compiler.version == "4.9":
+                        for version in ("4.8", "4.7", "4.6"):
+                            compatible_pkg = CompatiblePackage(self)
+                            compatible_pkg.settings.compiler.version = version
+                            self.compatible_packages.append(compatible_pkg)
+            """)
+        profile = textwrap.dedent("""
+            [settings]
+            os = Linux
+            compiler=gcc
+            compiler.version=4.9
+            compiler.libcxx=libstdc++
+            """)
+        client.save({"conanfile.py": conanfile,
+                     "myprofile": profile})
+        client.run("create . pkg/0.1@user/stable -pr=myprofile -s compiler.version=4.8")
+        self.assertIn("pkg/0.1@user/stable: Package '22c594d7fed4994c59a1eacb24ff6ff48bc5c51c'"
+                      " created", client.out)
+
+        client.save({"conanfile.py": GenConanfile().with_require_plain("pkg/0.1@user/stable")})
+        client.run("install . -pr=myprofile")
+        self.assertIn("pkg/0.1@user/stable:22c594d7fed4994c59a1eacb24ff6ff48bc5c51c", client.out)
+        self.assertIn("pkg/0.1@user/stable: Already installed!", client.out)
+
+    def compatible_option_test(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile, CompatiblePackage
+
+            class Pkg(ConanFile):
+                options = {"optimized": [1, 2, 3]}
+                default_options = {"optimized": 1}
+                def package_id(self):
+                    for optimized in range(int(self.options.optimized), 0, -1):
+                        compatible_pkg = CompatiblePackage(self)
+                        compatible_pkg.options.optimized = optimized
+                        self.compatible_packages.append(compatible_pkg)
+            """)
+        client.save({"conanfile.py": conanfile})
+        client.run("create . pkg/0.1@user/stable")
+        self.assertIn("pkg/0.1@user/stable: Package 'a97db2488658dd582a070ba8b6c6975eb1601a33'"
+                      " created", client.out)
+
+        client.save({"conanfile.py": GenConanfile().with_require_plain("pkg/0.1@user/stable")})
+        client.run("install . -o pkg:optimized=2")
+        # Information messages
+        self.assertIn("pkg/0.1@user/stable: ERROR: Compatible package ID "
+                      "d97fb97a840e4ac3b5e7bb8f79c87f1d333a85bc equal to the default package ID",
+                      client.out)
```

Comment:
But that is not possible. Lets say that you create a package for gcc 4.9 packageID1, and you define that it is unconditionally compatible with gcc 4.8 packageID2. This is all good.

Now you install with gcc 4.8, it will try to resolve packageID2, it will not be there (as we haven't created it), and it will start checking the fallback IDs, and as it was not conditional, it will go and check again the packageID2 id, which obviously will not be there. I am now outputting an error, but continuing, not failing, just skipping. Maybe change it to a warning only? Or doing nothing?

Reply:
I'm thinking about people that might be looking strictly to the output and checking for any `ERROR` to fail. Also, I think there was an issue asking for an environment variable to turn WARNINGS into actual errors :S

Reply:
We can do a ``info`` message too. Wdyt @lasote?

Reply:
Do an info message.